builds/e2e/upgrade-scenarios.yaml

```diff
@@ -0,0 +1,229 @@
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.0"
+          runtimeModulesVersion: "release/1.3"
+        1.1.8_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.8"
+          runtimeModulesVersion: "release/1.3"
+        1.1.13_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.1.13"
+          runtimeModulesVersion: "release/1.3"
+        1.2.0_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "release/1.3"
+        1.2.9_to_1.3_no_bootstrap_agent:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "release/1.3"
+
+        # Scenarios 6 to 10 test the 1.3 Edge Daemon with older versions of the 
+        # bootstrap Edge Agent, and runtime modules. These represent scenarios where
+        # the Edge Daemon is upgraded, but the config.toml/config.yaml and deployment 
+        # are left using older runtime images.
+        1.2.0_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.0"
+          runtimeModulesVersion: "1.2.0"
+        1.2.9_to_1.3_only_edge_daemon:
+          edgeDaemonVersion: "release/1.3"
+          bootstrapAgentVersion: "1.2.9"
+          runtimeModulesVersion: "1.2.9"
+        # TODO: Uncomment if/when the .NET test framework is updated to allow overriding
+        # the default 1.2 schema version used in the EH desired properties section
+        # of the deployment file. Until then, these scenarios will fail
+        # 1.1.0_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.0"
+        #   runtimeModulesVersion: "1.1.0"
+        # 1.1.8_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.8"
+        #   runtimeModulesVersion: "1.1.8"
+        # 1.1.13_to_1.3_only_edge_daemon:
+        #   edgeDaemonVersion: "release/1.3"
+        #   bootstrapAgentVersion: "1.1.13"
+        #   runtimeModulesVersion: "1.1.13"
+
+    timeoutInMinutes: 90
+
+    steps:
+    - template: templates/e2e-setup.yaml
+
+    # If edgeDaemonVersion != "release/1.3", remove the aziot-edge and
+    # aziot-identity-service packages from $(System.ArtifactsDirectory) and download the desired
+    # versions.
+    - bash: |
+        if [ $(edgeDaemonVersion) != "release/1.3" ]; then
+          pushd $(System.ArtifactsDirectory)/$(artifactName)
+          rm aziot-edge*.deb
+          rm aziot-identity-service*.deb
+          apt-get download aziot-edge='$(edgeDaemonVersion)-1'
+          is_version=`sudo apt depends aziot-edge='$(edgeDaemonVersion)-1' | sed -En "s/Depends: aziot-identity-service \(= ([0-9]+.[0-9]+.[0-9]+-[0-9]+)\)/\1/p"`
```

Comment:
Will this routine handle say.. `~` or `-` in case of RC correctly?
i.e. 1.2.0~rc1

Reply:
Good idea but in the interest of time and simplicity let's leave this for a future PR if/when we need it.